File: SOLUTIONS/LEETCODE/MEDIUM/longestSubarray/longestSubarray.js
function longestSubarray(nums){
    // get the maximum number from the subarray
    const maxi = Math.max(...nums)
    // initialise a counter to check repetitions of maxi
    let counter = 0
    // initialise longest to store the most repetition of maxi as you run through the list
    let longest = 0

    // loop through the nums array
    for(let i = 0; i < nums.length; i++){
        // if iterable is same as highest number
        if(nums[i] === maxi){
            // increment count
            counter++
            // if the current iterable stored in count is has more repetitions than longest
            if(counter > longest){
                // counter becomes the new longest
                longest = counter
            }
        // no need to store the number
        } else {counter = 0}
    }
    // return the max integer in the subarray with the longest repetition of itself
    return longest
}

File: SOLUTIONS/LEETCODE/MEDIUM/longestSubarray/longestSubarray.py
def longestSubarray(nums):
    maxi = max(nums)
    counter = 0
    longest = 0

    for i in range(len(nums)):
        if nums[i] == maxi:
            counter += 1
            if counter > longest:
                longest = counter
        else:
            counter = 0
    
    return longest

# The syntax, methods and thought process used above is the same with the JavaScript version of the solution